src/code_08026044.c

```diff
@@ -436,3 +436,583 @@ void sub_08026FAC(struct Cutscene *arg0) {
+        arg0->unk35C = 1;
+    }
+    arg0->unkF4.y = arg0->unk340 >> 8;
+    arg0->unk11C.y = arg0->unkF4.y;
+    if (arg0->unk340 == 0x8c00) {
+        SpriteInitNoTilesVram(&arg0->unkF4, 0x300, 0x390, 0xe, 0, 0xff, 0x10, 5, arg0->unk33C >> 8, arg0->unk340 >> 8, 0x42000);
+        arg0->flags &= ~0x10000;
+        SpriteInitNoTilesVram(&arg0->unk11C, 0x300, 0x391, 0xe, 0, 0xff, 0x10, 0xf, arg0->unkF4.x, arg0->unkF4.y, 0x42000);
+        arg0->flags &= ~0x300000;
+        arg0->flags |= 0x80000;
+        m4aSongNumStart(SE_08D5EDC4);
+        arg0->unk35A = 0;
+        arg0->unk0 = sub_0802A110;
+    }
+}
+
+void sub_0802A138(struct Cutscene *arg0);
+void sub_080279F4(struct Cutscene *arg0) {
+    struct Sprite *spr = &arg0->unkF4;
+    SpriteInitNoTilesVram(spr, 0x300, 0x390, 0x16, 0, 0xff, 0x10, 5, spr->x, spr->y, 0x42000);
+    arg0->flags &= ~0x30000;
+    SpriteInitNoTilesVram(&arg0->unk11C, 0x300, 0x391, 0x16, 0, 0xff, 0x10, 0xf, arg0->unkF4.x, arg0->unkF4.y, 0x42000);
+    arg0->flags &= ~0x300000;
+    arg0->flags |= 0x80000;
+    m4aSongNumStart(SE_08D5EDE4);
+    arg0->unk35A = 0;
+    arg0->unk0 = sub_0802A138;
+}
+
+void sub_08027BB8(struct Cutscene *arg0);
+void sub_08027AC0(struct Cutscene *arg0) {
+    struct Sprite *spr;
+    arg0->flags |= 0x110002;
+    arg0->flags &= ~0x220004;
+    spr = &arg0->unk4[0];
+    SpriteInitNoTilesVram(spr, 0x300, 0xb, 0, 0, 0xff, 0x10, 0, spr->x, spr->y, 0x2000);
+    gMainFlags |= 0x30000;
+    InvertRgbMap();
+    if (gMainFlags & MAIN_FLAG_BG_PALETTE_TRANSFORMATION_ENABLE) {
+        LoadBgPaletteWithTransformation(gBgPalette, 0, sizeof(gBgPalette) / 2);
+    }
+    else {
+        DmaCopy16(3, gBgPalette, gBgPalette, sizeof(gBgPalette));
+        gMainFlags |= MAIN_FLAG_BG_PALETTE_SYNC_ENABLE;
+    }
+    if (gMainFlags & MAIN_FLAG_OBJ_PALETTE_TRANSFORMATION_ENABLE) {
+        LoadObjPaletteWithTransformation(gObjPalette, 0, sizeof(gObjPalette) / 2);
+    }
+    else {
+        DmaCopy16(3, gObjPalette, gObjPalette, sizeof(gObjPalette));
+        gMainFlags |= MAIN_FLAG_OBJ_PALETTE_SYNC_ENABLE;
+    }
+    arg0->unk35A = 0;
+    arg0->unk0 = sub_08027BB8;
+}
+
+void sub_0802A160(struct Cutscene *arg0);
+void sub_08027BB8(struct Cutscene *arg0) {
+    if (arg0->unk35A++ > 10) {
+        if (gMainFlags & MAIN_FLAG_BG_PALETTE_TRANSFORMATION_ENABLE) {
+            LoadBgPaletteWithTransformation(gBgPalette, 0, sizeof(gBgPalette) / 2);
+        }
+        else {
+            DmaCopy16(3, gBgPalette, gBgPalette, sizeof(gBgPalette));
+            gMainFlags |= MAIN_FLAG_BG_PALETTE_SYNC_ENABLE;
+        }
+        if (gMainFlags & MAIN_FLAG_OBJ_PALETTE_TRANSFORMATION_ENABLE) {
+            LoadObjPaletteWithTransformation(gObjPalette, 0, sizeof(gObjPalette) / 2);
+        }
+        else {
+            DmaCopy16(3, gObjPalette, gObjPalette, sizeof(gObjPalette));
+            gMainFlags |= MAIN_FLAG_OBJ_PALETTE_SYNC_ENABLE;
+        }
+        CpuCopy32(arg0->unk214, gRgbMap, sizeof(gRgbMap));
```

Comment:
`CpuCopy32` demands src/dst to be 32bit-aligned so in this case we should add an alignment attribute to `unk214` and `gRgbMap`.

Reply:
Done! (I used `aligned(4)`)